src/main/java/com/Elessar/app/ClientMain.java

```diff
@@ -1,45 +1,105 @@
 package com.Elessar.app;
 
 import com.Elessar.app.client.MyClient;
-import java.io.IOException;
-import java.net.URL;
+import com.Elessar.app.client.MyClientServer;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.net.InetAddress;
+import java.net.NetworkInterface;
+import java.util.Collections;
+import java.util.Enumeration;
 
 /**
  * Created by Hans on 1/13/19.
  */
 public class ClientMain {
     public static void main(String[] args){
```

Comment:
as we discussed, we need a map<String, List<String>> member in clientMain and pass it to ClientServer.

Reply:
There are two questions puzzles me:
1. Initially we thought to use map for the mapping between toUser and message list, but since each clientMain corresponds to one single user, I think just using list should be sufficient, is it true?
2. Once we get the map or list whatever, how should we print it in clientMain? I think we need to print the message immediately once we receive it. But if we are in the middle of something, how could we jump to the print statement instantly after message arrives?

Reply:
1. Yes, a list should be enough.
2. You can change the loop in main function such that at the beginning of each iteration you first check if the list is empty. If not print all messages in the list and then do what we are doing right now.

Reply:
For 2, I think of this, but for example, the clientMain is waiting for user to type an option and in the mean time, new message comes in, then it won't get printed until user types option and this option finishes.

Reply:
If you really want to display the messages as soon as possible then I guess one way is like this:
before we execute any action user asked, let's first check if any message has arrived, if yes print the messages and ask user if he still wants to perform the action. If yes perform the action otherwise run the loop again and let user pick an action